python/lsst/ts/Dome/mock_llc/mock_motion/elevation_motion.py

```diff
@@ -0,0 +1,150 @@
+# This file is part of ts_Dome.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ElevationMotion"]
+
+import logging
+import math
+
+from .base_llc_motion import BaseLlcMotion
+from lsst.ts.idl.enums.Dome import MotionState
+import lsst.ts.salobj as salobj
+
+
+class ElevationMotion(BaseLlcMotion):
+    """Simulator for the elevation motion of the light and wind screen of the
+    Dome.
+
+    Parameters
+    ----------
+    start_position: `float`
+        The initial position [rad].
+    min_position: `float`
+        The minimum allowed position [rad].
+    max_position: `float`
+        The maximum allowed position [rad].
+    max_speed: `float`
+        The maximum allowed speed [rad/s].
+    start_tai: `float`
+        The current TAI time.
+
+    Notes
+    -----
+    This simulator can either move the light and wind screen to a target
+    position at maximum speed and stop, or crawl at the specified velocity. It
+    handles the 0 and 1/2pi radians (0 and 90 degrees) boundaries by stopping
+    there. When either moving or crawling, a new move or crawl command is
+    handled and the elevation motion/crawl can be stopped. To "park" the light
+    and wind screen, it needs to be moved to position 0.
+
+    """
+
+    def __init__(
+        self, start_position, min_position, max_position, max_speed, start_tai
+    ):
+        super().__init__(
+            start_position=start_position,
+            min_position=min_position,
+            max_position=max_position,
+            max_speed=max_speed,
+            start_tai=start_tai,
+        )
+        self.log = logging.getLogger("MockPointToPointActuator")
+
+    def get_position_and_motion_state(self, tai):
```

Comment:
Please consider also returning the velocity (for this and the azimuth actuator). Surely you will report velocity as well as position?

Reply:
Thanks for catching that. I meant to add that already but it must have slipped my mind. Done.